inputs/make_metadata/00_extract_srp_ids.R

```diff
@@ -0,0 +1,48 @@
+library(dplyr)
+library(readr)
+setwd("~/github/2022-microberna/")
```

Comment:
This assumes that any future user will store the root directory in a folder called github. Using a package like [here](https://cran.r-project.org/web/packages/here/index.html) or [this stackoverflow answer](https://stackoverflow.com/a/55322344/10930590) would let you `setwd` the root of the directory without losing portability

Reply:
ya i always cringe a little when i do this. When I write scripts that are executed by snakemake, the working directory path is dealt with properly because snakemake uses the directory that the snakefile is in as the working directory. This is also more seamlessly dealt with in notebooks, which set the current working directory to the directory of the notebook when opened with jupyter (and since jupyter is such a specific engine, there isn't much risk that the notebook will be opened in anything other than jupyter).

Usually, the scripts that I write that don't have to be re-executed by other users, that I run in RStudio, and that I still want to document, I use the `setwd()` approach because it's easy. 

looking at the links you posted above, I could put this at the top of my scripts instead:

```
library(tidyverse)
getCurrentFileLocation <-  function()
{
    this_file <- commandArgs() %>% 
    tibble::enframe(name = NULL) %>%
    tidyr::separate(col=value, into=c("key", "value"), sep="=", fill='right') %>%
    dplyr::filter(key == "--file") %>%
    dplyr::pull(value)
    if (length(this_file)==0)
    {
      this_file <- rstudioapi::getSourceEditorContext()$path
    }
    return(dirname(this_file))
}
setwd(dirname(getCurrentFileLocation()))
library(here)
setwd(here())
```

This defines a function that gets the path of the current file. It uses that path to set the working directory. The `here` package then traverses up until it sees a `.git` file, and records the path of that directory. When then use that path to set the working directory a last time.

I can't put the `getCurrentFileLocation` in the `scripts/utils.R`, because I wouldn't be able to source that script without knowing the proper path. So unfortunately, it would be a lot of code at the top of each script. this would also only work if the file is opened in the RStudio API. 

Thoughts? I'm sort of leaning toward either leaving it as is, or putting the code into a notebook, because this approach is verbose and hard to read/figure out what's happening.

Reply:
Yeah that makes sense. There aren't any clean/clear R solutions that I know of, so if no one else is going to run the script it is probably better to leave it as is

Reply:
ok thanks Ben! I hadn't seen the `here()` library or `rstudioapi::getSourceEditorContext()$path` before...it's good to know they exist!